### PR TITLE
Add support to mute unneeded Artisan commands from summary

### LINKED ISSUE #5: A feature request to "mute" unused commands?
Hi! Not sure if Issues is the right place for feature requests, but let's try. I came across this package and it immediately became a must-have on my list - Artisan looks a little bit cleaner.

What I'm suggesting may also clean it up a lot. What if we had a way to optionally hide certain commands from the list via config file? Would be nice to get rid of some default commands on the list while still keeping them working. Is it technically possible?

### PR DESCRIPTION
**Pull request by librevlad**
I guess this closes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,28 @@ Require Laravel Console Summary using [Composer](https://getcomposer.org):
...
+```
+
+You may want to clean up the summary screen by hiding some well-known commands from it. They will still be available in your application.
+Supports wildcards:
+
+```php
+// config/laravel-console-summary.php
+
+return [
+    'muted_commands' => [
```
Comment (nunomaduro) [inline README.md]:
```suggestion
    'hide' => [
```

File: config/config.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+/**
```
Comment (nunomaduro) [inline config/config.php]: Remove this comments.

File: config/config.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+ * Console Summary Configuration
+ */
+
+return [
+    'muted_commands' => [
```
Comment (nunomaduro) [inline config/config.php]:
```suggestion
    'hide' => [
```

Comment (nunomaduro) [inline config/config.php]: Can you add here something like this type of comments:

```php
    /*
    |--------------------------------------------------------------------------
    | Authentication Defaults
    |--------------------------------------------------------------------------
    |
    | This option controls the default authentication "guard" and password
    | reset options for your application. You may change these defaults
    | as required, but they're a perfect start for most applications.
    |
    */
```

File: src/Describer.php [hunk 4]
```diff
@@ -83,8 +83,15 @@ protected function describeCommands(Application $application, OutputInterface $o
     {
         $this->width = 0;
 
+        $muted = collect(config('laravel-console-summary.muted_commands'));
```
Comment (nunomaduro) [inline src/Describer.php]:
```suggestion
        $hide = collect(config('laravel-console-summary.hide'));
```

Comment (nunomaduro) [reply src/Describer.php]: What happens if the config don't exist?

File: src/Describer.php [hunk 5]
```diff
@@ -83,8 +83,15 @@ protected function describeCommands(Application $application, OutputInterface $o
...
+
         $namespaces = collect($application->all())->filter(function ($command) {
             return ! $command->isHidden();
+        })->filter(function($command) use($muted){
```
Comment (nunomaduro) [inline src/Describer.php]:
```suggestion
        })->filter(function($command) use($hide){
```

File: src/Describer.php [hunk 6]
```diff
@@ -83,8 +83,15 @@ protected function describeCommands(Application $application, OutputInterface $o
...
+            $nameParts = explode(':', $name = $command->getName());
+            $has_exact_match = $muted->contains( $command->getName() ) ;
```
Comment (nunomaduro) [inline src/Describer.php]: Violation to PSR1: Coding Style. Some goes for other parts of the code.

**Review by nunomaduro — CHANGES_REQUESTED** [review]

**Comment by nunomaduro** [comment]
Great job so far!

File: src/Describer.php (hunk 4 above)
Comment (librevlad) [reply src/Describer.php]: From the Service Provider:

```php
$this->mergeConfigFrom( __DIR__ . '/../config/config.php', 'laravel-console-summary' );
```

**Comment by librevlad** [comment]
Thank you! Adjusted those things. Should I somehow squash those commits now? Literally doing all this stuff for the first time :)

**Comment by nunomaduro** [comment]
Don't worry about the commit messages, at the end we squash them.

**Comment by librevlad** [comment]
@nunomaduro can you please review the changes and let me know if there's anything else to be done? Thanks!

**Comment by nunomaduro** [comment]
Thanks!

**Pull request merged**